### ISSUE TITLE
How to introduce openai triton in candle?

### ISSUE THREAD

**Issue by bofen97** [huggingface/candle#1463]
The handwritten CUDA operator is very complicated. How can we use openai triton in candle to simplify this process. ：）

**Comment by LaurentMazare** [comment]
Triton can be used to generate PTX code and candle uses PTX for its current kernels (except that they are generated using the nvcc compiler rather than triton) - see for example [this file](https://github.com/huggingface/candle/blob/main/candle-examples/examples/custom-ops/cuda_kernels.rs). It shoud be pretty straightforward to hook triton generated ptx at the same place, what would be good is someone writing some tutorial material to help users doing this.

**Comment by bofen97** [comment]
Yes, I will try to do it. At present, the documentation of candle needs to be improved.

**Comment by jeromeku** [comment]
@LaurentMazare @bofen97
Happy to work on this as well.  Been working on codegen for Triton kernels that support frameworks other than PyTorch.  See [here](https://github.com/openai/triton/pull/2845) and [here](https://github.com/openai/triton/pull/2642).

**Comment by jeromeku** [comment]
@LaurentMazare @bofen97 

Created a [minimal example](https://github.com/jeromeku/triton-rs) of loading a Triton kernel in Rust.

Lots to be done to make interfacing with `Triton` generated kernels more ergonomic per the notes in the aforementioned repo.
Let me know this is something of interest -- happy to explain / add more detailed examples.

**Comment by bofen97** [comment]
> minimal example

Thanks, I will study your code.
